Xamarin.Essentials/Contacts/Contacts.shared.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Contacts
+    {
+        public static async Task<Contact> PickContactAsync()
+        {
+            if (DeviceInfo.Platform != DevicePlatform.iOS)
+            {
+                var permissionStatus = await Permissions.RequestAsync<Permissions.ContactsRead>();
+                if (permissionStatus != PermissionStatus.Granted)
+                    return null;
+            }
```

Comment:
What happens on iOS when you call this?

DO we maybe want to throw if there is no permissions? I think we usually use `EnsureGrantedAsync` instead and the dev can catch and handle. I think null means canceled?

Reply:
I was thinking of denying permissions as being cancelled here.  I guess we should review what we do for things like the MediaPicker when permission is denied...

Reply:
iOS BTW does not require permissions for the picker: https://developer.apple.com/documentation/contactsui/cncontactpickerviewcontroller

Looking at MediaPicker, we do throw if not granted (EnsureDeclared) will change this for consistency.